### PR TITLE
Fixed NodeMainExecutorService and added dependency of fixed rosjava_c…

### LINKED ISSUE #254: NodeMainExecutorService leaks RosActivity through NodeMainExecutorServiceListener
When a RosActivity is destroyed, and the main application is still running, the NodeMainExecutorService contains a reference to the RosActivity preventing it from gc. The fix is to add the ability to remove the NodeMainExecutorServiceListener created in the RosActivity from the NodeMainExecutorService ListenerGroup. This requires ListenerGroup to contain a method for removing a listener (bug filed in rosjava/rosjava_core https://github.com/rosjava/rosjava_core/issues/233).

### PR DESCRIPTION
**Pull request by stratomda**
This fixes #254. This however has a dependency on rosjava/rosjava_core#233 and requires package a bumped package for rosjava_core. I assumed package rosjava_core:0.2.2.

### CODE REVIEW

**Comment by adamantivm** [comment]
@stratomda any chance we can get this targeted for kinetc?

File: android_10/src/org/ros/android/RosActivity.java [hunk 1]
```diff
@@ -84,7 +87,15 @@ public void onShutdown(NodeMainExecutorService nodeMainExecutorService) {
...
     public void onServiceDisconnected(ComponentName name) {
+      nodeMainExecutorService.removeListener(serviceListener);
+      serviceListener = null;
     }
+
+    public NodeMainExecutorServiceListener getServiceListener()
+    {
+      return serviceListener;
+    }
+
```
Comment (adamantivm) [inline android_10/src/org/ros/android/RosActivity.java]: really tiny nit: please remove extra new line here

**Review by adamantivm — APPROVED** [review]
This is a great contribution @stratomda thank you. I only left a very minor comment that I hope you can address. Also, it would be better if we could target to kinetic instead of indigo - or at least in addition to indigo.

**Comment by stratomda** [comment]
@adamantivm I can merge this into kinetic branch. You'll have to give me a few days to get my workspace spun up. I needed this for a specific project using indigo, but I am using my own maven repo, so no need to target indigo.

**Comment by adamantivm** [comment]
@stratomda I released the rosjava_core fix for indigo and cherry picked for kinetic. We could do the same for this fix as well. Could you address that tiny nit and then I can take care of the rest?

**Comment by stratomda** [comment]
Will do. Thanks for all the work you do.

**Comment by stratomda** [comment]
@adamantivm removed extra line.

**Comment by adamantivm** [comment]
JFYI, this and rosjava/rosjava_core#233 are merged, tagged and released to the rosjava maven repo.

**Pull request merged**